src/overlays/actors/ovl_Obj_Tokei_Tobira/z_obj_tokei_tobira.c

```diff
@@ -29,23 +30,140 @@ ActorInit Obj_Tokei_Tobira_InitVars = {
+CollisionHeader* D_80ABD770[] = { &object_tokei_tobira_Colheader_0012B0, &object_tokei_tobira_Colheader_001590 };
+f32 D_80ABD778[] = { 1.0f, -1.0f };
+Gfx* D_80ABD780[] = { object_tokei_tobira_DL_001108, object_tokei_tobira_DL_0013E8 };
 
-extern InitChainEntry D_80ABD750[];
+void ObjTokeiTobira_Init(Actor* thisx, PlayState* play) {
+    ObjTokeiTobira* this = THIS;
+    s32 pad;
+    s32 type = OBJTOKEITOBIRA_GET_TYPE(&this->dyna.actor);
+    Vec3f pos;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tokei_Tobira/ObjTokeiTobira_Init.s")
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    DynaPolyActor_Init(&this->dyna, DYNA_TRANSFORM_POS | DYNA_TRANSFORM_ROT_Y);
+    DynaPolyActor_LoadMesh(play, &this->dyna, D_80ABD770[type]);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tokei_Tobira/ObjTokeiTobira_Destroy.s")
+    if (type == OBJTOKEITOBIRA_TYPE_0) {
+        Actor_SpawnAsChild(&play->actorCtx, &this->dyna.actor, play, ACTOR_OBJ_TOKEI_TOBIRA,
+                           this->dyna.actor.world.pos.x, this->dyna.actor.world.pos.y, this->dyna.actor.world.pos.z,
+                           this->dyna.actor.shape.rot.x, this->dyna.actor.shape.rot.y, this->dyna.actor.shape.rot.z,
+                           OBJTOKEITOBIRA_PARAM(OBJTOKEITOBIRA_TYPE_1));
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tokei_Tobira/func_80ABD3B0.s")
+    Matrix_RotateYS(D_80ABD76C[type] + this->dyna.actor.shape.rot.y, MTXMODE_NEW);
+    Matrix_MultVec3f(&D_80ABD760, &pos);
+    this->dyna.actor.world.pos.x += pos.x;
+    this->dyna.actor.world.pos.y += pos.y;
+    this->dyna.actor.world.pos.z += pos.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tokei_Tobira/ObjTokeiTobira_Update.s")
+    if ((type == OBJTOKEITOBIRA_TYPE_0) && !(CHECK_WEEKEVENTREG(WEEKEVENTREG_59_04)) &&
+        (play->sceneId == SCENE_CLOCKTOWER) && (gSaveContext.sceneLayer == 0) && (this->dyna.actor.csId >= 0)) {
+        this->dyna.actor.flags |= ACTOR_FLAG_10;
+        this->actionFunc = ObjTokeiTobira_StartCutscene;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Tokei_Tobira/ObjTokeiTobira_Draw.s")
+void ObjTokeiTobira_Destroy(Actor* thisx, PlayState* play) {
+    ObjTokeiTobira* this = THIS;
+
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+}
+
+void ObjTokeiTobira_StartCutscene(ObjTokeiTobira* this) {
+    if (CutsceneManager_IsNext(this->dyna.actor.csId)) {
+        CutsceneManager_StartWithPlayerCs(this->dyna.actor.csId, &this->dyna.actor);
+        SET_WEEKEVENTREG(WEEKEVENTREG_59_04);
+        this->actionFunc = NULL;
+        this->dyna.actor.flags &= ~ACTOR_FLAG_10;
+    } else {
+        CutsceneManager_Queue(this->dyna.actor.csId);
+    }
+}
+
+void ObjTokeiTobira_Update(Actor* thisx, PlayState* play) {
+    s32 pad;
+    ObjTokeiTobira* this = THIS;
+    Player* player = GET_PLAYER(play);
+    s32 pad50;
+    s32 temp_param = OBJTOKEITOBIRA_GET_TYPE(&this->dyna.actor);
```

Comment:
You can rename this as `type`.
```suggestion
    s32 type = OBJTOKEITOBIRA_GET_TYPE(&this->dyna.actor);
```
Don't forget to rename all the uses